### ISSUE TITLE
Don't rely on the live api for the tests

### ISSUE THREAD

**Issue by jnioche** [Boavizta/cloud-scanner#645]
## Bug description

see #643

Our tests can fail because we are hitting the live version of the API on https://api.boavizta.org/ and its version can change anytime.

1.  start a Docker instance with a version known to be compatible within the tests
   https://rust.testcontainers.org/quickstart/testcontainers/ is probably the best solution for this

2. mock what the API returns 
   https://docs.rs/wiremock/latest/wiremock/

**Comment by demeringo** [comment]
Yes, this subject will become even more important when we start to integrate multiple cloud providers and need to test the inventory / scanning part (calling provider API's). 

The current "test" implementation of the inventory is very naive, it calls the real AWS API, so it needs to have real resouces in the cloud to produce results. This is not portable.
